lib/metadataTypes/Campaign.js

```diff
@@ -21,31 +21,129 @@ class Campaign extends MetadataType {
      * @returns {Promise.<TYPE.MetadataTypeMapObj>} Promise
      */
     static async retrieve(retrieveDir, _, __, key) {
-        const res = await super.retrieveREST(retrieveDir, '/hub/v1/campaigns', null, key);
+        this.upgradeGetBulk();
+        const res = await super.retrieveREST(
+            retrieveDir,
+            '/legacy/v1/beta2/data/campaign/',
+            null,
+            key
+        );
         // get assignments
-
+        Util.logger.info(`Retrieving: campaignAsset`);
         const campaignAssets = await Promise.all(
             Object.keys(res.metadata).map((key) =>
-                this.getAssetTags(retrieveDir, res.metadata[key].id, key)
+                this.getAssetTags(retrieveDir, res.metadata[key].campaignId, key)
             )
         );
         Util.logger.info(
-            `Downloaded: campaignAssets (${campaignAssets.flat().length})` + Util.getKeysString(key)
+            `Downloaded: campaignAsset (${campaignAssets.flat().length})` + Util.getKeysString(key)
         );
 
         return res;
     }
+    /**
+     * Retrieves event definition metadata for caching
+     *
+     * @returns {Promise.<TYPE.MetadataTypeMapObj>} Promise of metadata
+     */
+    static retrieveForCache() {
+        this.upgradeGetBulk();
+        return super.retrieveREST(null, '/legacy/v1/beta2/data/campaign/');
+    }
+    /**
+     * helper for {@link MobileMessage.retrieve} and {@link MobileMessage.retrieveForCache}
+     */
+    static upgradeGetBulk() {
```

Comment:
ready for merge in sdk - lets use that instead

Reply:
of course - it's in my to-dos :) just waiting to get it released with you :)